File: client-app/src/app/models/blogPost.ts
export interface IBlogPost {
    id: string;
    title: string;
    date: Date;
}

export class BlogPost implements IBlogPost {
    constructor(
        id: string,
        title: string,
        date: Date) {
        this.id = id;
        this.title = title;
        this.date = date;
    }
    id: string;
    title: string;
    date: Date;
}

export interface IBlogPostDetail extends IBlogPost {
    content: string;
}

export class BlogPostDetail extends BlogPost implements IBlogPostDetail {
    constructor(
        id: string,
        title: string,
        date: Date,
        content: string) {
        super(id, title, date);
        this.content = content;
    }
    content: string;
}

File: PersonalWebsite/Domain/BlogPost.cs
﻿namespace Domain
{
    public class BlogPost
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public string Content { get; set; }
        public DateTime Date { get; set; }
    }
}
